File: generate_seed.py
import json
import random
import uuid
from datetime import datetime

# Creator identities to ensure consistency
creators = [
    {"username": "Alex Chen", "avatar": "https://ui-avatars.com/api/?name=Alex+Chen&background=000&color=fff"},
    {"username": "Maya Patel", "avatar": "https://ui-avatars.com/api/?name=Maya+Patel&background=000&color=fff"},
    {"username": "Arjun Dev", "avatar": "https://ui-avatars.com/api/?name=Arjun+Dev&background=000&color=fff"},
    {"username": "Noah Williams", "avatar": "https://ui-avatars.com/api/?name=Noah+Williams&background=000&color=fff"},
    {"username": "Sarah Kim", "avatar": "https://ui-avatars.com/api/?name=Sarah+Kim&background=000&color=fff"},
    {"username": "Daniel Thomas", "avatar": "https://ui-avatars.com/api/?name=Daniel+Thomas&background=000&color=fff"},
    {"username": "Priya Nair", "avatar": "https://ui-avatars.com/api/?name=Priya+Nair&background=000&color=fff"},
    {"username": "Ethan Cole", "avatar": "https://ui-avatars.com/api/?name=Ethan+Cole&background=000&color=fff"},
    {"username": "Olivia Martin", "avatar": "https://ui-avatars.com/api/?name=Olivia+Martin&background=000&color=fff"},
    {"username": "Ryan Lee", "avatar": "https://ui-avatars.com/api/?name=Ryan+Lee&background=000&color=fff"},
]

# We need to create these profiles first if they don't exist
# But for the simulation, we'll map them to the existing creator_id if we have to, 
# or generate new UUIDs if the migration allows.
# Given Lovable constraints, we should use the existing user_id for now or create new profile rows.

def generate_resource(index, existing_creator_id):
    creator = creators[index % len(creators)]
    
    # Categories and types from schema
    categories = ["Website Development", "Mobile Applications", "UI/UX", "AI/ML", "Data Science", "Automation", "Marketing", "Content Creation", "Business", "Education", "Productivity"]
    types = ["Prompt", "Code", "UI/UX", "Component", "Template", "Project", "Module", "Function"]

    # 1. Modern Responsive Navbar
    if index == 0:
        return {
            "title": "Minimal Responsive Navbar",
            "type": "UI/UX",
            "category": "UI/UX",
            "description": "# Minimal Responsive Navbar\n\nA production-ready responsive navigation bar built with semantic HTML, modern CSS, and vanilla JavaScript. Features smooth transitions, mobile-first design, and accessibility best-practices.\n\n### Features\n- Desktop Mega-menu support\n- Mobile Hamburger Drawer\n- Sticky Header with Glassmorphism\n- Active Link detection\n- Keyboard Accessibility (Tab-friendly)\n\n### Installation\nSimply copy the HTML/CSS/JS files into your project and link them in your `<head>`.\n\n### Usage\nCustomize the `nav-links` array in `script.js` to match your site structure.",
            "content": "<!-- index.html -->\n<nav class='navbar'>\n  <div class='logo'>Promptly</div>\n  <ul class='nav-links'>\n    <li><a href='#'>Home</a></li>\n    <li><a href='#'>Explore</a></li>\n    <li><a href='#'>Resources</a></li>\n  </ul>\n  <button class='mobile-toggle'>☰</button>\n</nav>",
            "preview_url": "https://images.unsplash.com/photo-1507238691740-187a5b1d37b8?auto=format&fit=crop&q=80&w=800",
            "tags": ["navbar", "responsive", "navigation", "ui-ux"],
            "likes_count": random.randint(150, 500),
            "saves_count": random.randint(40, 120),
            "creator_id": existing_creator_id
        }
    
    # 2. Dashboard Sidebar
    elif index == 1:
        return {
            "title": "Modern SaaS Dashboard Sidebar",
            "type": "Component",
            "category": "UI/UX",
            "description": "A sleek, collapsible sidebar component for enterprise SaaS applications. Built with React and Tailwind CSS.\n\n### Features\n- Collapsible state management\n- Tooltip integration\n- Active route highlighting\n- Profile footer section\n- Dark mode compatible",
            "content": "// Sidebar.jsx\nimport React, { useState } from 'react';\nimport { Home, Settings, Users, BarChart } from 'lucide-react';\n\nconst Sidebar = () => {\n  const [collapsed, setCollapsed] = useState(false);\n  return (\n    <div className={`h-screen bg-white border-r transition-all ${collapsed ? 'w-20' : 'w-64'}`}>\n      {/* Navigation items */}\n    </div>\n  );\n};",
            "preview_url": "https://images.unsplash.com/photo-1460925895917-afdab827c52f?auto=format&fit=crop&q=80&w=800",
            "tags": ["dashboard", "sidebar", "saas", "react"],
            "likes_count": random.randint(200, 600),
            "saves_count": random.randint(80, 200),
            "creator_id": existing_creator_id
        }

    # 3. Authentication UI
    elif index == 2:
        return {
            "title": "Modern Authentication UI",
            "type": "Project",
            "category": "Website Development",
            "description": "Complete authentication frontend flows including Login, Signup, and Password Reset screens.\n\n### Included Views\n1. Login with Social Auth options\n2. Multi-step Signup\n3. Verification Email landing\n4. Password Recovery",
            "content": "// App.tsx\nimport { Login } from './pages/Login';\nimport { Signup } from './pages/Signup';\n\nexport const AuthStack = () => (\n  <Routes>\n    <Route path='/login' element={<Login />} />\n    <Route path='/signup' element={<Signup />} />\n  </Routes>\n);",
            "preview_url": "https://images.unsplash.com/photo-1555066931-4365d14bab8c?auto=format&fit=crop&q=80&w=800",
            "tags": ["auth", "login", "signup", "typescript"],
            "likes_count": random.randint(300, 800),
            "saves_count": random.randint(150, 400),
            "creator_id": existing_creator_id
        }

    # 4. AI Chat Interface
    elif index == 3:
        return {
            "title": "Minimal AI Chat Interface",
            "type": "Component",
            "category": "AI/ML",
            "description": "A ChatGPT-inspired chat interface with auto-scrolling, code highlighting, and streaming message simulation.\n\n### Features\n- Markdown rendering support\n- Code block syntax highlighting\n- Response streaming UI\n- Mobile responsive input area",
            "content": "// Chat.tsx\nconst Chat = () => {\n  const [messages, setMessages] = useState([]);\n  return (\n    <div className='flex flex-col h-full'>\n      <MessageList messages={messages} />\n      <ChatInput onSend={(m) => setMessages([...messages, m])} />\n    </div>\n  );\n};",
            "preview_url": "https://images.unsplash.com/photo-1676299081847-c0326a0333d5?auto=format&fit=crop&q=80&w=800",
            "tags": ["ai", "chat", "interface", "openai"],
            "likes_count": random.randint(400, 1000),
            "saves_count": random.randint(200, 500),
            "creator_id": existing_creator_id
        }

    # Default fallback for others during the loop
    return {
        "title": f"Resource {index+1}: {random.choice(['Advanced', 'Clean', 'Professional', 'Modern'])} {random.choice(['Library', 'Tool', 'Framework', 'Dashboard'])}",
        "type": random.choice(types),
        "category": random.choice(categories),
        "description": f"This is a high-quality resource created by {creator['username']}. It includes production-ready code and complete documentation.",
        "content": f"// Source code for resource {index+1}\nconsole.log('Hello Promptly');",
        "preview_url": f"https://picsum.photos/seed/{index}/800/600",
        "tags": ["sample", "resource", creator['username'].split()[0].lower()],
        "likes_count": random.randint(50, 300),
        "saves_count": random.randint(20, 100),
        "creator_id": existing_creator_id
    }

# This script generates a SQL migration to seed the data
# We'll use the creator_id we found earlier: 03dbcee7-522c-4865-8119-d59eef27319d

CREATOR_ID = "03dbcee7-522c-4865-8119-d59eef27319d"
resources_sql = []

for i in range(30):
    res = generate_resource(i, CREATOR_ID)
    title = res['title'].replace("'", "''")
    desc = res['description'].replace("'", "''")
    content = res['content'].replace("'", "''")
    tags = "{" + ",".join(f'"{t}"' for t in res['tags']) + "}"
    
    sql = f"INSERT INTO public.resources (title, type, category, description, content, preview_url, tags, creator_id, likes_count, saves_count) VALUES ('{title}', '{res['type']}', '{res['category']}', '{desc}', '{content}', '{res['preview_url']}', '{tags}', '{res['creator_id']}', {res['likes_count']}, {res['saves_count']});"
    resources_sql.append(sql)

print("\n".join(resources_sql))

File: scripts/generate-seed.js
import fs from 'fs';
import crypto from 'crypto';

function uuid() {
  return crypto.randomUUID();
}

// Generate 10 Creator Profiles
const creators = [
  { id: uuid(), username: 'alexc', full_name: 'Alex Chen', avatar_url: 'https://images.unsplash.com/photo-1535713875002-d1d0cf377fde?w=150&h=150&fit=crop' },
  { id: uuid(), username: 'mayap', full_name: 'Maya Patel', avatar_url: 'https://images.unsplash.com/photo-1494790108377-be9c29b29330?w=150&h=150&fit=crop' },
  { id: uuid(), username: 'arjund', full_name: 'Arjun Dev', avatar_url: 'https://images.unsplash.com/photo-1599566150163-29194dcaad36?w=150&h=150&fit=crop' },
  { id: uuid(), username: 'noahw', full_name: 'Noah Williams', avatar_url: 'https://images.unsplash.com/photo-1527980965255-d3b416303d12?w=150&h=150&fit=crop' },
  { id: uuid(), username: 'sarahk', full_name: 'Sarah Kim', avatar_url: 'https://images.unsplash.com/photo-1438761681033-6461ffad8d80?w=150&h=150&fit=crop' },
  { id: uuid(), username: 'danielt', full_name: 'Daniel Thomas', avatar_url: 'https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?w=150&h=150&fit=crop' },
  { id: uuid(), username: 'priyan', full_name: 'Priya Nair', avatar_url: 'https://images.unsplash.com/photo-1544005313-94ddf0286df2?w=150&h=150&fit=crop' },
  { id: uuid(), username: 'ethanc', full_name: 'Ethan Cole', avatar_url: 'https://images.unsplash.com/photo-1507003211169-0a1dd7228f2d?w=150&h=150&fit=crop' },
  { id: uuid(), username: 'oliviam', full_name: 'Olivia Martin', avatar_url: 'https://images.unsplash.com/photo-1534528741775-53994a69daeb?w=150&h=150&fit=crop' },
  { id: uuid(), username: 'ryanl', full_name: 'Ryan Lee', avatar_url: 'https://images.unsplash.com/photo-1500648767791-00dcc994a43e?w=150&h=150&fit=crop' }
];

const escapeSql = (str) => {
  if (!str) return 'NULL';
  return "'" + str.replace(/'/g, "''") + "'";
};

let sql = "-- ==========================================\n" +
"-- PROMPTLY SEED DATA\n" +
"-- ==========================================\n\n" +
"DELETE FROM public.comments;\n" +
"DELETE FROM public.likes;\n" +
"DELETE FROM public.saves;\n" +
"DELETE FROM public.ratings;\n" +
"DELETE FROM public.resources;\n\n";

sql += "-- 2. Insert Profiles\n";
for (const c of creators) {
  sql += "INSERT INTO public.profiles (id, username, full_name, avatar_url, onboarding_completed, role) VALUES (" + escapeSql(c.id) + ", " + escapeSql(c.username) + ", " + escapeSql(c.full_name) + ", " + escapeSql(c.avatar_url) + ", true, 'user') ON CONFLICT (id) DO UPDATE SET username = EXCLUDED.username, full_name = EXCLUDED.full_name, avatar_url = EXCLUDED.avatar_url;\n";
}

const resources = [];

const topics = [
  { t: "Minimal Responsive Navbar", c: "UI/UX", img: "https://images.unsplash.com/photo-1498050108023-c5249f4df085?auto=format&fit=crop&w=800&q=80" },
  { t: "Modern SaaS Dashboard Sidebar", c: "UI/UX", img: "https://images.unsplash.com/photo-1551288049-bebda4e38f71?auto=format&fit=crop&w=800&q=80" },
  { t: "Modern Authentication UI", c: "Website Development", img: "https://images.unsplash.com/photo-1555421689-491a97ff2040?auto=format&fit=crop&w=800&q=80" },
  { t: "Minimal AI Chat Interface", c: "AI/ML", img: "https://images.unsplash.com/photo-1677442136019-21780ecad995?auto=format&fit=crop&w=800&q=80" },
  { t: "AI Prompt Enhancer", c: "AI/ML", img: "https://images.unsplash.com/photo-1620712943543-bcc4688e7485?auto=format&fit=crop&w=800&q=80" },
  { t: "Premium SaaS Landing Page", c: "Website Development", img: "https://images.unsplash.com/photo-1460925895917-afdab827c52f?auto=format&fit=crop&w=800&q=80" },
  { t: "Interactive E-commerce Product Card", c: "UI/UX", img: "https://images.unsplash.com/photo-1472851294608-062f824d29cc?auto=format&fit=crop&w=800&q=80" },
  { t: "Modern Mobile Bottom Navigation", c: "Mobile Applications", img: "https://images.unsplash.com/photo-1512941937669-90a1b58e7e9c?auto=format&fit=crop&w=800&q=80" },
  { t: "Premium Flutter Login Screen", c: "Mobile Applications", img: "https://images.unsplash.com/photo-1526498460520-4c246339dccb?auto=format&fit=crop&w=800&q=80" },
  { t: "Python Data Analytics Dashboard", c: "Data Science", img: "https://images.unsplash.com/photo-1543286386-2e659306cd6c?auto=format&fit=crop&w=800&q=80" },
  { t: "AI Image Generation Prompts", c: "AI/ML", img: "https://images.unsplash.com/photo-1682687982501-1e5898cb8e4b?auto=format&fit=crop&w=800&q=80" },
  { t: "Tailwind UI Components", c: "UI/UX", img: "https://images.unsplash.com/photo-1507238692062-58751aa51417?auto=format&fit=crop&w=800&q=80" },
  { t: "Production Ready REST API", c: "Website Development", img: "https://images.unsplash.com/photo-1555066931-4365d14bab8c?auto=format&fit=crop&w=800&q=80" },
  { t: "JWT Authentication System", c: "Website Development", img: "https://images.unsplash.com/photo-1618401471353-b98afee0b2eb?auto=format&fit=crop&w=800&q=80" },
  { t: "Modern Admin Panel", c: "Website Development", img: "https://images.unsplash.com/photo-1551009175-8a68da93d5f9?auto=format&fit=crop&w=800&q=80" },
  { t: "Cinematic Animated Hero", c: "UI/UX", img: "https://images.unsplash.com/photo-1618005182384-a83a8bd57fbe?auto=format&fit=crop&w=800&q=80" },
  { t: "Advanced File Upload UI", c: "UI/UX", img: "https://images.unsplash.com/photo-1611162617474-5b21e879e113?auto=format&fit=crop&w=800&q=80" },
  { t: "Developer Markdown Editor", c: "Productivity", img: "https://images.unsplash.com/photo-1542831371-29b0f74f9713?auto=format&fit=crop&w=800&q=80" },
  { t: "Python File Organizer script", c: "Automation", img: "https://images.unsplash.com/photo-1526374965328-7f61d4dc18c5?auto=format&fit=crop&w=800&q=80" },
  { t: "Python Web Scraper Tool", c: "Automation", img: "https://images.unsplash.com/photo-1555099962-4199c345e5dd?auto=format&fit=crop&w=800&q=80" },
  { t: "Minimal React Design System", c: "UI/UX", img: "https://images.unsplash.com/photo-1561070791-2526d30994b5?auto=format&fit=crop&w=800&q=80" },
  { t: "SaaS Database Schema Model", c: "Data Science", img: "https://images.unsplash.com/photo-1633356122544-f134324a6cee?auto=format&fit=crop&w=800&q=80" },
  { t: "Full Stack Docker Env", c: "Website Development", img: "https://images.unsplash.com/photo-1605745341112-85968b19335b?auto=format&fit=crop&w=800&q=80" },
  { t: "Production CI/CD Actions", c: "Automation", img: "https://images.unsplash.com/photo-1581291518857-4e27b48ff24e?auto=format&fit=crop&w=800&q=80" },
  { t: "Website AI Assistant Bot", c: "AI/ML", img: "https://images.unsplash.com/photo-1587620962725-abab7fe55159?auto=format&fit=crop&w=800&q=80" },
  { t: "Developer Portfolio Template", c: "Website Development", img: "https://images.unsplash.com/photo-1454165804606-c3d57bc86b40?auto=format&fit=crop&w=800&q=80" },
  { t: "UX Component Generation Prompts", c: "UI/UX", img: "https://images.unsplash.com/photo-1557821552-17105153ce9a?auto=format&fit=crop&w=800&q=80" },
  { t: "Mobile Finance App Design", c: "Mobile Applications", img: "https://images.unsplash.com/photo-1607799279861-4dddf8b60dd5?auto=format&fit=crop&w=800&q=80" },
  { t: "AI Document Analyzer NLP", c: "AI/ML", img: "https://images.unsplash.com/photo-1504868584819-f8e8b4b6d7e3?auto=format&fit=crop&w=800&q=80" },
  { t: "Full Stack Next.js SaaS", c: "Website Development", img: "https://images.unsplash.com/photo-1537432376769-00f5c2f4c8d2?auto=format&fit=crop&w=800&q=80" }
];

function createResource(title, category, type, description, contentObj, readme, preview_url) {
  const creator = creators[Math.floor(Math.random() * creators.length)];
  return {
    id: uuid(),
    title,
    category,
    type,
    description,
    content: JSON.stringify(contentObj),
    readme_content: readme,
    preview_url,
    creator_id: creator.id,
    difficulty: ['Beginner', 'Intermediate', 'Advanced'][Math.floor(Math.random() * 3)],
    language: 'JavaScript',
    framework: 'React',
    license_type: 'MIT',
    likes_count: Math.floor(Math.random() * 2000) + 50,
    saves_count: Math.floor(Math.random() * 1000) + 10,
    downloads_count: Math.floor(Math.random() * 3000) + 100,
    views_count: Math.floor(Math.random() * 10000) + 500,
    tags: JSON.stringify(['Modern', 'Responsive', 'Clean']),
  };
}

for (const t of topics) {
  const codeContent = "// " + t.t + "\nimport React from 'react';\n\nexport default function App() {\n  return <div>" + t.t + "</div>;\n}\n";

  resources.push(createResource(
    t.t,
    t.c,
    "Project",
    "A premium quality implementation of " + t.t + ". Ready for production use with best practices, responsive design, beautiful micro-interactions, and clean code architecture.",
    {
      files: [
        { filename: "App.jsx", extension: ".jsx", language: "JavaScript", content: codeContent },
        { filename: "package.json", extension: ".json", language: "JSON", content: "{\"name\":\"project\"}" }
      ]
    },
    "# " + t.t + "\n\n## Overview\nThis project provides a complete, polished solution for **" + t.t + "**.\n\n## License\nMIT",
    t.img
  ));
}

sql += "\n-- 3. Insert Resources\n";
for (const r of resources) {
  sql += "INSERT INTO public.resources (id, title, category, type, description, content, readme_content, preview_url, creator_id, difficulty, language, framework, license_type, likes_count, saves_count, downloads_count, views_count, tags) VALUES (" + escapeSql(r.id) + ", " + escapeSql(r.title) + ", " + escapeSql(r.category) + ", " + escapeSql(r.type) + ", " + escapeSql(r.description) + ", " + escapeSql(r.content) + ", " + escapeSql(r.readme_content) + ", " + escapeSql(r.preview_url) + ", " + escapeSql(r.creator_id) + ", " + escapeSql(r.difficulty) + ", " + escapeSql(r.language) + ", " + escapeSql(r.framework) + ", " + escapeSql(r.license_type) + ", " + r.likes_count + ", " + r.saves_count + ", " + r.downloads_count + ", " + r.views_count + ", " + escapeSql(r.tags) + ");\n";
}

const timestamp = "20260808235959";
const filename = "supabase/migrations/" + timestamp + "_seed_real_data.sql";

fs.writeFileSync(filename, sql, 'utf-8');
console.log('Successfully generated migration file: ' + filename);
